[Monitoring] Only look at ES for the missing data alert for now

chrisronline opened the pull request:
Relates to https://github.com/elastic/kibana/issues/83309

Until we can come up with a strategy to detect upgrades versus legitimate downed stack products, we are changing the missing monitoring data alert to only apply to Elasticsearch, as upgrades to Elasticsearch typically persist node ids.

To test, I'd recommend creating a cloud deployment on a version that is less than the current one. Configure your local Kibana's monitoring settings to read monitoring data from this deployment. Then, upgrade the cloud deployment and ensure we don't see any missing monitoring data alerts.

chrisronline committed:
Only look at ES for the missing data alert for now

elasticmachine commented:
Pinging @elastic/stack-monitoring (Team:Monitoring)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/)
* Commit: a496312fa831fe108f90f16d3dc20bcd000c6648
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup8](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/execution/node/651/log)
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/execution/node/603/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/","number":"88608","commit":"a496312fa831fe108f90f16d3dc20bcd000c6648"}]}
PIPELINE-->

chrisronline commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into monitoring/missing_data_only_es

qn895 committed:
[ML] Persist URL state for Anomaly detection jobs using metric function (#83507)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mshustov committed:
Update typescript eslint to v4.8 (#83520)

* update deps

* update rules

use type-aware @typescript-eslint/no-shadow instead of no-shadow. do not use no-undef, rely on TypeScript instead

* fix or mute all lint errors

* react-hooks eslint plugin fails on ? syntax

* fix wrong typings in viz

* remove React as a global type

* fix eslint errors

* update version to 4.8.1

* fix a new error

afharo committed:
[Telemetry] Move Monitoring collection strategy to a collector (#82638)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

ravikesarwani commented:
@chrisronline I am thinking that we should modify the "looking back" period for this alert from 1 day to 6 hours.
This will reduce the noise as we still are learning the full behavior of this alert in different scenarios.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88624/)
* Commit: 4331211356c0a4efddd413a2a9ec1736bea78b22

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #88608](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/) failed a496312fa831fe108f90f16d3dc20bcd000c6648

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88608/","number":"88608","commit":"a496312fa831fe108f90f16d3dc20bcd000c6648"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/88624/","number":"88624","commit":"4331211356c0a4efddd413a2a9ec1736bea78b22"}]}
PIPELINE-->

watson committed:
Bump is-my-json-valid to v2.20.5 (#83642)

watson committed:
Bump jsonpointer to v4.1.0 (#83641)

watson committed:
Bump y18n@5 to v5.0.5 (#83644)

watson committed:
Bump flat to v4.1.1 (#83647)

dgieselaar committed:
[APM] Improve router types (#83620)

* [APM] Improve router types

* Pass processorEvent param to useDynamicIndexPattern

scottybollinger committed:
[Workplace Search] Port Box changes from ent-search (#83675)

scottybollinger committed:
[Workplace Search] Migrate SourceLogic from ent-search (#83593)

* Initial copy/paste of source logic

Only changed lodash imports and import order for linting

* Add types and route

* Update paths and typings

Renamed IMeta -> Meta
Used object instead of IObject

* Remove internal flash messages in favor of globals

- All instances of flashAPIErrors(e) are only placeholders until the later commit removing axios.

- buttonLoading was set to false when the error flash messages were set. For now I added a `setButtonNotLoading` action to do this manually in a finally block. This will be refactored once axios is removed.

- SourcesLogic is no longer needed because we set a queued flash message instead of trying to set it in SourcesLogic, which no longer has local flash messages

* Add return types to callback definitions

* Update routes

According to the API info getSourceReConnectData is supposed to send the source ID and not the service type. In the template, we are actually sending the ID but the logic file parameterizes it as serviceType. This is fixed here.

Usage: https://github.com/elastic/ent-search/blob/master/app/javascript/workplace_search/ContentSources/components/AddSource/ReAuthenticate.tsx#L38

* Replace axios with HttpLogic

Also removes using history in favor of KibanaLogic’s navigateToUrl

* Fix incorrect type

This selector is actually an array of strings

* Create GenericObject to satisfy TypeScript

Previously in `ent-search`, we had a generic `IObject` interface that we could use on keyed objects. It was not migrated over since it uses `any` and Kibana has a generic `object` type we can use in most situations. However, when we are checking for keys in our code, `object` does not work. This commit is an attempt at making a generic interface we can use.

* More strict object typing

Removes GenericObject from last commit and adds stricter local typing

* Add i18n

Also added for already-merged SourcesLogic

* Move button loading action to finally block

* Move route strings to inline

mshustov committed:
disable incremenetal build for legacy tsconfig.json (#82986)

ymao1 committed:
Not resetting server log level if level is defined (#83651)

ymao1 committed:
[Alerting] Add `alert.updatedAt` field to represent date of last user edit (#83578)

* Adding alert.updatedAt field that only updates on user edit

* Updating unit tests

* Functional tests

* Updating alert attributes excluded from AAD

* Fixing test

* PR comments

YulNaumenko committed:
Fixed console error, which appears when saving changes in Edit Alert flyout (#83610)

mistic committed:
fix(NA): search examples kibana version declaration (#83182)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[Maps] Add query bar inputs to geo threshold alerts tracked points & boundaries (#80871)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

yuliacech committed:
Add Managed label to data streams and a view switch for the table (#83049)

* Add Managed label to data streams and a view switch for the table

* Fix i18n errors

* Updated some wording and made filter function easier (managed data streams)

* Update x-pack/plugins/index_management/__jest__/client_integration/home/data_streams_tab.test.ts

Co-authored-by: Alison Goryachev <alisonmllr20@gmail.com>

* Renamed view to include (managed data streams)

* Update x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_list.tsx

Co-authored-by: James Rodewig <40268737+jrodewig@users.noreply.github.com>

* Update x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_list.tsx

Co-authored-by: James Rodewig <40268737+jrodewig@users.noreply.github.com>

* Update x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_table/data_stream_table.tsx

Co-authored-by: James Rodewig <40268737+jrodewig@users.noreply.github.com>

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Alison Goryachev <alisonmllr20@gmail.com>
Co-authored-by: James Rodewig <40268737+jrodewig@users.noreply.github.com>

kobelb committed:
Updating code-owners to use new core/app-services team names (#83731)

* Updating code-owners to use new core/app-services team names

* And the comment as well

Commit:
[Maps] Add 'crossed' & 'exited' events to tracking alert (#82463)

Commit:
[Enterprise Search] Rename React Router helpers (#83718)

* Rename EUI React Router components

- Instead of bogarting the EUI component names, use EuiLinkTo instead of EuiLink

Other misc renaming
- eui_link.tsx to eui_components.tsx for clearer file name
- EuiReactRouterHelper to ReactRouterHelper, to make the distinction between EUI and React Router clearer (in theory you could use this helper for non-EUI components)
- other misc type renaming

* Update simple instances of previous EUI RR components to Eui*To

* Clean up complex/renamed instances of Eui*To
(hopefully much more straightforward now)

- unfortunately side_nav requires an eslint disable

scottybollinger committed:
[Workplace Search] Update SourceIcon to match latest changes in ent-search (#83714)

* Move source icons into subfolder

* Copy over new icons

* Update SourceIcon to account for full bleed images

* Remove unused file

* Fix broken icon path

Commit:
[App Search] Engine overview layout stub (#83504)

* Set up Overview file

* Finish Overview page logic, stub out empty/metric views

* Stub in basic empty engine overview
- Minus document creation button & API code example

* Stub out EngineOverviewMetrics and unavailable empty prompt

* Stub out EngineOverMetrics components (stats, charts, logs)

* [Refactor] Pull out some document creation i18n strings to constants

- They're repeated/reused by the DocumentCreationPopover component

* PR feedback: Drop the regex

* PR feedback: RecentLogs -> RecentApiLogs

* PR feedback: Copy

* PR feedback: Copy, sentence-casing

mistic committed:
chore(NA): update lmdb store to v0.8.15 (#83726)

* chore(NA): upgrade lmdb-store to v0.8.15

* chore(NA): remove unused ts-error statements

simianhacker committed:
[Metrics UI] Optimizations for Snapshot and Inventory Metadata (#83596)

* [Metrics UI] Add time range to inventory metadata request

* Adding optimizations for snapshot request

* Adding sorting to dataset request

* Only query inventory metadata for AWS

* moving check inside getCloudMetadata

* removing unused deps

bmcconaghy committed:
Adding documentation for global action configuration options (#83557)

* Adding documentation for global action configuration options

* Update docs/user/alerting/defining-alerts.asciidoc

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

* incorporating PR feedback

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

spalger committed:
Revert "[App Search] Engine overview layout stub (#83504)"

This reverts commit 71f972dc837d6186ad2b8157af8fb178d86e6a96.

nreese committed:
[maps] convert VectorStyleEditor to TS (#83582)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

lukasolson committed:
[data.search] Server-side background session service (#81099)

* [Search] Add request context and asScoped pattern

* Update docs

* Unify interface for getting search client

* [WIP] [data.search] Server-side background session service

* Update examples/search_examples/server/my_strategy.ts

Co-authored-by: Anton Dosov <dosantappdev@gmail.com>

* Review feedback

* Fix checks

* Add tapFirst and additional props for session

* Fix CI

* Fix security search

* Fix test

* Fix test for reals

* Add restore method

* Add code to search examples

* Add restore and search using restored ID

* Fix handling of preference and order of params

* Trim & cleanup

* Fix types

* Review feedback

* Add tests and remove handling of username

* Update docs

* Move utils to server

* Review feedback

* More review feedback

* Regenerate docs

* Review feedback

* Doc changes

Co-authored-by: Anton Dosov <dosantappdev@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

pheyos committed:
Increase bulk request timeout during esArchiver load (#83657)

This PR fixes some timeouts during esArchive load by increasing the request timeout.

tylersmalley committed:
Skip failing cypress test

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

dgieselaar committed:
Make expectSnapshot available in all functional test runs (#82932)

Co-authored-by: spalger <spalger@users.noreply.github.com>

cnasikas committed:
[Security Solution][Detections] Fix adding an action to detection rules (#83722)

ymao1 committed:
Revert "[Alerting] Add `alert.updatedAt` field to represent date of last user edit (#83578)"

This reverts commit acc3e2f443e3c60dfc923aa1b3b179f34cf69804.

Commit:
Forward any registry cache-control header for files (#83680)

closes #83631 

### Problem
Assets are served with a `cache-control` header that prevents any caching
<img src="https://user-images.githubusercontent.com/640/99534379-517d2300-2975-11eb-8c05-4fb3f127c52b.png"/>

### Root cause
Likely from this code https://github.com/elastic/kibana/blob/2a365ff6329544465227e61141ded6fba8bb2c80/src/core/server/http/http_tools.ts#L40-L43

Also based on these tests, it seems this is default/expected behavior

https://github.com/elastic/kibana/blob/b3eefb97da8e712789b5c5d2eeae65c886ed8f64/src/core/server/http/integration_tests/router.test.ts#L510-L520

### Proposed solution
Set the header via the response handler as shown in this test:
https://github.com/elastic/kibana/blob/b3eefb97da8e712789b5c5d2eeae65c886ed8f64/src/core/server/http/integration_tests/router.test.ts#L522-L536

### This PR
If this registry response contains a `cache-control` header, that value is included in the EPM response as well

In `master`, which points to `epr-snapshot`
<img width="742" alt="Screen Shot 2020-11-18 at 12 33 47 PM" src="https://user-images.githubusercontent.com/57655/99568352-4fc75580-299d-11eb-962f-6ff28fa9510d.png">
which matches https://epr-snapshot.elastic.co/package/apache/0.2.6/img/logo_apache.svg

or using `epr.elastic.co`, 
<img width="781" alt="Screen Shot 2020-11-18 at 12 31 56 PM" src="https://user-images.githubusercontent.com/57655/99568350-4fc75580-299d-11eb-966e-f3489c13edb5.png">
which matches https://epr.elastic.co/package/apache/0.2.6/img/logo_apache.svg

gmmorris committed:
fixed pagination in connectors list (#83638)

Ensures we specify the page on the EuiTable so that pagination is retain after rerenders.

nchaulet committed:
[Fleet] Rename ingestManager plugin ID fleet (#83200)

mattkime committed:
[index patterns] improve index pattern cache (#83368)

* cache index pattern promise, not index pattern

gmmorris committed:
[Task Manager] Ensures retries are inferred from the schedule of recurring tasks (#83682)

This addresses a bug in Task Manager in the task timeout behaviour. When a recurring task's `retryAt` field is set (which happens at task run), it is currently scheduled to the task definition's `timeout` value, but the original intention was for these tasks to retry on their next scheduled run (originally identified as part of https://github.com/elastic/kibana/issues/39349).

In this PR we ensure recurring task retries are scheduled according to their recurring schedule, rather than the default `timeout` of the task type.

mistic committed:
skip flaky suite (#83793)

mistic committed:
skip flaky suite (#65278)

mistic committed:
skip flaky suite (#83771)

mistic committed:
skip flaky suite (#83773)

dgieselaar committed:
Improve snapshot error messages (#83785)

Commit:
[Data Table] Remove extra column in split mode (#83193)

* Fix extra column in split table

* Update table exports

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
test just part of the message to avoid updates (#83703)

jgowdyelastic committed:
[ML] Space management UI (#83320)

* [ML] Space management UI

* fixing types

* small react refactor

* adding repair toasts

* text and style changes

* handling spaces being disabled

* correcting initalizing endpoint response

* text updates

* text updates

* fixing spaces manager use when spaces is disabled

* more text updates

* switching to delete saved object first rather than overwrite

* filtering non ml spaces

* renaming file

* fixing types

* updating list style

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

KOTungseth committed:
[DOCS] Consolidates plugins (#83712)

Commit:
Disable exporting/importing of templates.  Optimize pitch images a bit (#83098)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[Enterprise Search] Engine overview layout stub (#83756)

* Set up Overview file

* Finish Overview page logic, stub out empty/metric views

* Stub in basic empty engine overview
- Minus document creation button & API code example

* Stub out EngineOverviewMetrics and unavailable empty prompt

* Stub out EngineOverMetrics components (stats, charts, logs)

* [Refactor] Pull out some document creation i18n strings to constants

- They're repeated/reused by the DocumentCreationPopover component

* PR feedback: Drop the regex

* PR feedback: RecentLogs -> RecentApiLogs

* PR feedback: Copy

* PR feedback: Copy, sentence-casing

* I forgot to rebase against my own PR :dead_inside:

igoristic reviewed (commented):
@chrisronline I think we should also optimize (remove none es logic) in: `x-pack/plugins/monitoring/server/lib/alerts/fetch_missing_monitoring_data.ts` and make it only look in `.monitoring-es-*` indices, which should make it significantly faster

KOTungseth committed:
[DOCS] Reallocates limitations to point-of-use (#79582)

* [DOCS] Reallocates limitations to point-of-use

* KQL changes

* Removed limitations file

* Review comments

mistic committed:
skip flaky suite (#79389)

pmuellr committed:
[alerts] adds action group and date to mustache template variables for actions (#83195)

resolves: https://github.com/elastic/kibana/issues/67389

Adds new variables to the existing set of variables that can be used in mustache templates to be used in action parameters when creating alerts.

- `alertActionGroup` - the action group associated with the alert scheduling actions
- `date` - the current date, in ISO format

jbudz committed:
skip "Dashboards linked by a drilldown are both copied to a space" (#83824)

chrisronline committed:
Fix small issue with detecting missing monitoring data from APM (#83646)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

chrisronline committed:
PR feedback

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

chrisronline commented:
Sorry folks, bad merge. Replaced by https://github.com/elastic/kibana/pull/83839
